### PR TITLE
Frontend: Fe structure setup

### PR DESCRIPTION
**Pull request by ArnavNath550**
1. frontend file structure added
2. navigation bar added
3. custom components
     a. buttons
     b. search input added 

## Summary by CodeRabbit

- **New Features**
	- Introduced a navigation bar with search functionality and project submission button.
	- Added a new `HomePage` component with routing via `react-router-dom`.
	- Implemented a customizable `Button` component with support for icons, tooltips, and loading states.
	- Created a `SearchInput` component for enhanced search capabilities.
	- Added a `Tooltip` component for attaching customizable tooltips to elements.
- **Style Updates**
	- Updated the application's primary and background colors, and introduced new color definitions in `tailwind.config.js`.
	- Major CSS adjustments including new `.gyan-container` class, updated `#root` element styling, and a radial gradient background for the body.
- **Refactor**
	- Removed old logo animations, media queries, and state management logic from `App.tsx`.
	- Replaced specific style properties with custom CSS variables in `index.css`.
- **Documentation**
	- Added type definitions for input fields and a new `SVGComponent` type for SVG elements in React.
- **Chores**
	- Set up a proxy configuration in `vite.config.ts` for API requests redirection.
- **New Libraries**
	- Utilized `react-router-dom` for app routing, `@radix-ui/react-tooltip` for tooltips, and `tailwind-merge` for CSS class name merging.

### CODE REVIEW

**Review by shravan20 — COMMENTED** [review]
Nice work!! Suggested few changes rest everything looks good. 🔥

File: apps/ui/src/components/navigation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import { GoIssueDraft, GoStack, GoProjectTemplate } from "react-icons/go";
+import { Button } from '../../packages/ui/components/buttons/Button';
+import { GiArrowCursor } from 'react-icons/gi';
+
+const Navigation: React.FC = () => {
+    return (
+        <div>
+            <div className="h-50 w-full bg-primary border-b-info border-b-[1px] absolute top-0 p-4 pt-2 pb-2 backdrop-blur-sm">
+                <div className="w-[1200px] m-auto z-50 relative flex flex-row justify-between h-full items-center">
```
[the remark this answers is not in the document]
Comment (ArnavNath550) [reply apps/ui/src/components/navigation/index.tsx:12]: nope nope, for mobile devices i have to create a breakpoint for that.. for now i've just set for 1200px to cover desktop devices

File: apps/ui/vite.config.ts.timestamp-1706528809724-1ba7b7dca2c65.mjs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
...ICAic291cmNlc0NvbnRlbnQiOiBbImNvbnN0IF9fdml0ZV9pbmplY3RlZF9vcmlnaW5hbF9kaXJuYW1lID0gXCIvVXNlcnMvYWRtaW4vRGVza3RvcC93ZWIvcHJvZGV4cGxvcmEvYXBwcy91aVwiO2NvbnN0IF9fdml0ZV9pbmplY3RlZF9vcmlnaW5hbF9maWxlbmFtZSA9IFwiL1VzZXJzL2FkbWluL0Rlc2t0b3Avd2ViL3Byb2RleHBsb3JhL2FwcHMvdWkvdml0ZS5jb25maWcudHNcIjtjb25zdCBfX3ZpdGVfaW5qZWN0ZWRfb3JpZ2luYWxfaW1wb3J0X21ldGFfdXJsID0gXCJmaWxlOi8vL1VzZXJzL2FkbWluL0Rlc2t0b3Avd2ViL3Byb2RleHBsb3JhL2FwcHMvdWkvdml0ZS5jb25maWcudHNcIjtpbXBvcnQgeyBkZWZpbmVDb25maWcgfSBmcm9tICd2aXRlJ1xuaW1wb3J0IHJlYWN0IGZyb20gJ0B2aXRlanMvcGx1Z2luLXJlYWN0LXN3YydcblxuLy8gaHR0cHM6Ly92aXRlanMuZGV2L2NvbmZpZy9cbmV4cG9ydCBkZWZhdWx0IGRlZmluZUNvbmZpZyh7XG4gIHBsdWdpbnM6IFtyZWFjdCgpXSxcbiAgc2VydmVyOiB7XG4gICAgcHJveHk6IHtcbiAgICAgIFwiL2FwaVwiOiB7XG4gICAgICAgIHRhcmdldDogXCJodHRwOi8vbG9jYWxob3N0OjMwMDBcIixcbiAgICAgICAgY2hhbmdlT3JpZ2luOiB0cnVlLFxuICAgICAgfSxcbiAgICB9LFxuICB9LFxufSlcbiJdLAogICJtYXBwaW5ncyI6ICI7QUFBc1QsU0FBUyxvQkFBb0I7QUFDblYsT0FBTyxXQUFXO0FBR2xCLElBQU8sc0JBQVEsYUFBYTtBQUFBLEVBQzFCLFNBQVMsQ0FBQyxNQUFNLENBQUM7QUFBQSxFQUNqQixRQUFRO0FBQUEsSUFDTixPQUFPO0FBQUEsTUFDTCxRQUFRO0FBQUEsUUFDTixRQUFRO0FBQUEsUUFDUixjQUFjO0FBQUEsTUFDaEI7QUFBQSxJQUNGO0FBQUEsRUFDRjtBQUNGLENBQUM7IiwKICAibmFtZXMiOiBbXQp9Cg==
```
[the remark this answers is not in the document]
Comment (ArnavNath550) [reply apps/ui/vite.config.ts.timestamp-1706528809724-1ba7b7dca2c65.mjs:18]: i'm not too sure.. i don't think we need to.. i must've missed seeing this file while committing

File: apps/ui/src/libs/classNames.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { twMerge } from "tailwind-merge";
+
+export default function classNames(...classes: unknown[]) {
+  return twMerge(classes.filter(Boolean).join(" "));
+}
```
Comment (shravan20) [inline apps/ui/src/libs/classNames.ts:5]:
```suggestion
}

```

File: apps/ui/src/components/navigation/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (shravan20) [reply apps/ui/src/components/navigation/index.tsx:12]: Ah, understood!

File: apps/ui/src/packages/ui/components/forms/inputs/type.d.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  containerClassName?: string;
+  showAsteriskIndicator?: boolean;
+  t?: (key: string) => string;
+  onClickAddon?: () => void;
+} & React.ComponentProps<typeof Input> & {
+    labelProps?: React.ComponentProps<typeof Label>;
+    labelClassName?: string;
+  };
+
+export type InputProps = JSX.IntrinsicElements["input"] & { isFullWidth?: boolean };
```
Comment (shravan20) [inline apps/ui/src/packages/ui/components/forms/inputs/type.d.ts:26]:
```suggestion
export type InputProps = JSX.IntrinsicElements["input"] & { isFullWidth?: boolean };

```

**Review by shravan20 — APPROVED** [review]
PR LGTM

**Pull request closed without merging**